peak/peak.py

```diff
@@ -66,7 +72,41 @@ def call_wrapper(*args,**kwargs):
         return call_wrapper
     return decorator
 
+#This decorator does the following:
+#1) Caches the function call
+#2) Stores the family closure in Peak._fc_
+def family_closure(f):
+    num_inputs = f.__code__.co_argcount
+    if num_inputs != 1:
+        raise ValueError("Family Closure must take a single input 'family'")
```

Comment:
Why? (I.e. why not wait till failure on line 85, if it is a failure)

Reply:
There has been a lot of code written passing in family and generator parameters at the same time. I am preventing that in user code. For example not having the check would not prevent the following:

```
def gen_PE(family, width=16):
   ...
```

Reply:
And?

Reply:
Not sure what you are questioning. The point of family_closure is to force users to adopt a single style for writing peak classes that allows the auto-mapper to work. This style involves only passing in a single family as an argument.

Reply:
What breaks if you allow `def gen_PE(family, width=16):`?   The width might become an effectively irrelevant argument but what's the issue?

Reply:
I am fine with a warning here but seems unnecessary to make it an error.

Reply:
Also you don't block `def gen_pe(family, *, width=16):`.  So I don't see why you should block `def gen_pe(family, width=16)`.

Reply:
Changed to Warning